### PR TITLE
WIP: ActionHero + Serverless

### PR DESCRIPTION
**Pull request by evantahler**
This PR is a WIP demonstrating some of the ideas about running ActionHero in a "serverless" (AWS Lambda, etc) environment. 

1. `npm install --global serverless`
2. checkout this branch
3. disable the servers we don't need (web, websocket) by changing `enabled = false` in their config files. 

Now, you can run your actions!

* Run an action: `serverless invoke local -f randomNumber`
* Run an action with params: `serverless invoke local -f cacheTest -l --data '{"key": "key", "value": "value"}'`
* Get a file (base-64 encoded) `serverless invoke local -f file -l --data '{"file": "simple.html"}'`

TODO:
- This runs locally on node.js v8+... but AWS only supports v6.  You *can* ship custom binaries with Lambda functions... we should figure out how.
- Is there a way to auto-generate `serverless.yml` from `api.actions.actions`?
- Tests... somehow.

### CODE REVIEW

**Comment by evantahler** [comment]
This is generally on hold until AWS Lambda supports node v8+

File: handler.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+const awaitRunning = async () => {
+  await sleep()
+  if (api.running !== true) {
+    return awaitRunning()
+  }
+}
+
+(async () => {
+  module.exports.run = async (event, context, callback) => {
+    await awaitRunning()
```
Comment (gcoonrod) [inline handler.js]: Is this a check to wait for the server to finish starting up?

File: servers/lambda.js [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+  goodbye () {}
+
+  async handleConnection (connection) {
+    if (typeof connection.rawConnection.event === 'object') {
+      connection.params = connection.rawConnection.event
+    }
+
+    const parts = connection.rawConnection.context.functionName.split('-')
+    const actionName = parts[(parts.length - 1)]
```
Comment (gcoonrod) [inline servers/lambda.js]: These two lines seem brittle...

File: handler.js (hunk 1 above)
Comment (evantahler) [reply handler.js]: Yeah... the weird thing here is that the "entrypoint" here is this exported function, so this method is the first thing that is called, *then* you need to boot actionhero :/

File: servers/lambda.js (hunk 2 above)
Comment (evantahler) [reply servers/lambda.js]: very! good call.

**Comment by chimmelb** [comment]
Woot! 
https://aws.amazon.com/blogs/compute/node-js-8-10-runtime-now-available-in-aws-lambda/

**Pull request closed without merging**